### PR TITLE
version bump on k8s io client-go and apimachinery

### LINKED ISSUE #13958: Incompatible with k8s.io/client-go@v0.27.0+
One of my other dependencies pulled in a transient update of `k8s.io/client-go` and my code started to fail to compile with the following error:

> `github.com/prysmaticlabs/prysm/v5@v5.0.3/beacon-chain/cache/common.go:10:23: cannot use popProcessNoopFunc (value of type func(_ interface{}) error) as "k8s.io/client-go/tools/cache".PopProcessFunc value in argument to queue.Pop`

Would it be possible to fix this incompatibility?

### PR DESCRIPTION
**Pull request by hopinheimer**
Fixes #13958

Screenshot 2024-09-11 at 3 51 46 PM

Apologies from my extra PR, it's just me being a neat freak cause I messed up the last one

also i noticed that the deps.bzl is getting tracked, is that alright?

### CODE REVIEW

**Comment by prestonvanloon** [comment]
@hopinheimer thanks! Yes, we need the deps.bzl updates here. Please run the bazel command and push the deps.bzl file.

**Comment by hopinheimer** [comment]
@prestonvanloon done! my pleasure

**Review by prestonvanloon — APPROVED** [review]

**Comment by prestonvanloon** [comment]
hi @hopinheimer, looks like this doesn't build. 

Please address the API changes / build issues

**Comment by hopinheimer** [comment]
No problemo! I'll look into it

**Comment by hopinheimer** [comment]
builds now! @prestonvanloon lmk if I need do some more tests

**Review by prestonvanloon — APPROVED** [review]

**Review by prestonvanloon — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ The format is based on Keep a Changelog, and this project adheres to Semantic Ve
 - Light client support: abstracted out the light client headers with different versions.
 - `ApplyToEveryValidator` has been changed to prevent misuse bugs, it takes a closure that takes a `ReadOnlyValidator` and returns a raw pointer to a `Validator`. 
 - Removed gorilla mux library and replaced it with net/http updates in go 1.22
-- Updated k8s-io/client-go to v0.30.4 and k8s-io/apimachinery to v0.30.4
```
Comment (prestonvanloon) [inline CHANGELOG.md]: Oops

**Pull request merged**